<?xml version="1.0" ?>
<sdf version="1.5">
  <model name="arm26">
    <muscles>model://arm26/muscles.osim</muscles>
    <link name="ground">
      <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
      <inertial>
        <mass>0</mass>
        <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
        <inertia>
          <ixx>0</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0</iyy>
          <iyz>0</iyz>
          <izz>0</izz>
        </inertia>
      </inertial>
    </link>
    <link name="base">
      <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
      <inertial>
        <mass>0</mass>
        <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
        <inertia>
          <ixx>0</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0</iyy>
          <iyz>0</iyz>
          <izz>0</izz>
        </inertia>
      </inertial>
      <visual name="ground_ribs">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://base/meshes/visual/ground_ribs.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="ground_spine">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://base/meshes/visual/ground_spine.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="ground_skull">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://base/meshes/visual/ground_skull.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="ground_jaw">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://base/meshes/visual/ground_jaw.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="ground_r_clavicle">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://base/meshes/visual/ground_r_clavicle.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="ground_r_scapula">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://base/meshes/visual/ground_r_scapula.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
    </link>
    <link name="r_humerus">
      <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
      <inertial>
        <mass>1.864572</mass>
        <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
        <inertia>
          <ixx>0.01481</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.004551</iyy>
          <iyz>0</iyz>
          <izz>0.013193</izz>
        </inertia>
      </inertial>
      <visual name="arm_r_humerus">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_humerus/meshes/visual/arm_r_humerus.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
    </link>
    <link name="r_ulna_radius_hand">
      <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
      <inertial>
        <mass>1.534315</mass>
        <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>
        <inertia>
          <ixx>0.019281</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.001571</iyy>
          <iyz>0</iyz>
          <izz>0.020062</izz>
        </inertia>
      </inertial>
      <visual name="arm_r_ulna">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_ulna.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_radius">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_radius.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_lunate">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_lunate.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_scaphoid">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_scaphoid.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_pisiform">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_pisiform.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_triquetrum">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_triquetrum.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_capitate">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_capitate.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_trapezium">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_trapezium.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_trapezoid">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_trapezoid.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_hamate">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_hamate.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_1mc">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_1mc.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_2mc">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_2mc.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_3mc">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_3mc.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_4mc">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_4mc.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_5mc">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_5mc.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_thumbprox">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_thumbprox.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_thumbdist">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_thumbdist.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_2proxph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_2proxph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_2midph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_2midph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_2distph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_2distph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_3proxph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_3proxph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_3midph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_3midph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_3distph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_3distph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_4proxph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_4proxph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_4midph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_4midph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_4distph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_4distph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_5proxph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_5proxph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_5midph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_5midph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
      <visual name="arm_r_5distph">
        <pose> -0 0 -0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://r_ulna_radius_hand/meshes/visual/arm_r_5distph.vtp</uri>
            <scale> 1 1 1</scale>
          </mesh>
        </geometry>
      </visual>
    </link>
    <joint>
      <child>ground</child>
    </joint>
    <joint name="offset">
      <parent>ground</parent>
      <child>base</child>
    </joint>
    <joint name="r_shoulder" type="MeDunnoYet">
      <parent>base</parent>
      <child>r_humerus</child>
    </joint>
    <joint name="r_elbow" type="MeDunnoYet">
      <parent>r_humerus</parent>
      <child>r_ulna_radius_hand</child>
    </joint>
    <plugin filename="libgazebo_ros_muscle_interface.so" name="muscle_interface_plugin"/>
  </model>
</sdf>
